fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/data/DelinquencyBucketData.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.delinquency.data;
+
+import java.io.Serializable;
+import java.util.List;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@ToString
+@Getter
+@Setter
+public class DelinquencyBucketData implements Serializable {
+
+    private Long id;
+    private String name;
+    private List<DelinquencyRangeData> ranges;
+
+    public DelinquencyBucketData(String name, List<DelinquencyRangeData> ranges) {
+        this.name = name;
+        this.ranges = ranges;
+    }
+
+    public DelinquencyBucketData(Long id, String name, List<DelinquencyRangeData> ranges) {
+        this.id = id;
+        this.name = name;
+        this.ranges = ranges;
+    }
+
+    public static DelinquencyBucketData instance(Long id, String name, List<DelinquencyRangeData> ranges) {
```

Comment:
Do you mind explaining why these static factories are needed? Thanks.

Reply:
Done! Removed